crates/forge-runner/src/trace_data.rs

```diff
@@ -26,310 +24,198 @@ pub const TRACE_DIR: &str = ".snfoundry_trace";
-            }
-            DeprecatedSyscallSelector::GetSequencerAddress => {
-                ProfilerDeprecatedSyscallSelector::GetSequencerAddress
-            }
-            DeprecatedSyscallSelector::GetTxInfo => ProfilerDeprecatedSyscallSelector::GetTxInfo,
-            DeprecatedSyscallSelector::GetTxSignature => {
-                ProfilerDeprecatedSyscallSelector::GetTxSignature
-            }
-            DeprecatedSyscallSelector::Keccak => ProfilerDeprecatedSyscallSelector::Keccak,
-            DeprecatedSyscallSelector::LibraryCall => {
-                ProfilerDeprecatedSyscallSelector::LibraryCall
-            }
-            DeprecatedSyscallSelector::LibraryCallL1Handler => {
-                ProfilerDeprecatedSyscallSelector::LibraryCallL1Handler
-            }
-            DeprecatedSyscallSelector::ReplaceClass => {
-                ProfilerDeprecatedSyscallSelector::ReplaceClass
-            }
-            DeprecatedSyscallSelector::Secp256k1Add => {
-                ProfilerDeprecatedSyscallSelector::Secp256k1Add
-            }
-            DeprecatedSyscallSelector::Secp256k1GetPointFromX => {
-                ProfilerDeprecatedSyscallSelector::Secp256k1GetPointFromX
-            }
-            DeprecatedSyscallSelector::Secp256k1GetXy => {
-                ProfilerDeprecatedSyscallSelector::Secp256k1GetXy
-            }
-            DeprecatedSyscallSelector::Secp256k1Mul => {
-                ProfilerDeprecatedSyscallSelector::Secp256k1Mul
-            }
-            DeprecatedSyscallSelector::Secp256k1New => {
-                ProfilerDeprecatedSyscallSelector::Secp256k1New
-            }
-            DeprecatedSyscallSelector::Secp256r1Add => {
-                ProfilerDeprecatedSyscallSelector::Secp256r1Add
-            }
-            DeprecatedSyscallSelector::Secp256r1GetPointFromX => {
-                ProfilerDeprecatedSyscallSelector::Secp256r1GetPointFromX
-            }
-            DeprecatedSyscallSelector::Secp256r1GetXy => {
-                ProfilerDeprecatedSyscallSelector::Secp256r1GetXy
-            }
-            DeprecatedSyscallSelector::Secp256r1Mul => {
-                ProfilerDeprecatedSyscallSelector::Secp256r1Mul
-            }
-            DeprecatedSyscallSelector::Secp256r1New => {
-                ProfilerDeprecatedSyscallSelector::Secp256r1New
-            }
-            DeprecatedSyscallSelector::SendMessageToL1 => {
-                ProfilerDeprecatedSyscallSelector::SendMessageToL1
-            }
-            DeprecatedSyscallSelector::StorageRead => {
-                ProfilerDeprecatedSyscallSelector::StorageRead
-            }
-            DeprecatedSyscallSelector::StorageWrite => {
-                ProfilerDeprecatedSyscallSelector::StorageWrite
-            }
-        }
+fn build_profiler_call_type(value: CallType) -> ProfilerCallType {
+    match value {
+        CallType::Call => ProfilerCallType::Call,
+        CallType::Delegate => ProfilerCallType::Delegate,
```

Comment:
What do you think about implementing `From` `CallType` trait for`ProfilerCalltype`?

Reply:
It would require profiler to depend on blockifier, which we want to avoid.

Reply:
Right, resolved then